examples/bank/bank.go

```diff
@@ -57,12 +57,19 @@ type Account struct {
 	Balance int64
 }
 
-func (a Account) encode() ([]byte, error) {
-	return json.Marshal(a)
+func (a Account) encode(retErr *error) []byte {
+	result, err := json.Marshal(a)
```

Comment:
you need to check retErr at the very beginning and exit if it's not nil, otherwise, you may end up overwriting the error as well as continuing to run code on parameters that are bad (due to previous errors)